test/helper.go

```diff
@@ -0,0 +1,277 @@
+					Namespace: TestNamespace,
+				},
+			},
+		},
+	}
+}
+
+func GotExpectedError(expected string, got error) bool {
+	if expected == "" {
+		return true
+	}
+	return strings.Contains(got.Error(), expected)
+}
+
+func GetTime() *v1.Time {
+	now := v1.Now()
+	return &now
+}
+
+func AssertNoErrorReconciliation() func(res ctrl.Result, err error, t *testing.T) {
+	return func(res ctrl.Result, err error, t *testing.T) {
+		if err != nil || !res.IsZero() {
+			t.Errorf("failed. Expected no error and empty response but got: err: %s, res: %v", err, res)
+		}
+	}
+}
+
+func GetGCTestRequest() ctrl.Request {
+	return ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Name: TestDummyCRName,
+		},
+	}
+}
+
+func GetValidTestClient(initLists ...client.ObjectList) client.WithWatch {
+	return fake.NewClientBuilder().
+		WithScheme(GetValidTestScheme()).
+		WithLists(initLists...).
+		Build()
+}
+
+func GetValidTestScheme() *runtime.Scheme {
+	scheme := runtime.NewScheme()
+	_ = v1beta1.AddToScheme(scheme)
+	_ = corev1.AddToScheme(scheme)
+	_ = v1alpha1.AddToScheme(scheme)
+	return scheme
+}
+
+func GetBasicScheme() *runtime.Scheme {
+	scheme := runtime.NewScheme()
+	_ = v1beta1.AddToScheme(scheme)
+	_ = corev1.AddToScheme(scheme)
+	return scheme
+}
+
+func VerifyGatewayClassAcceptance(name string, want bool) func(c client.Client, res ctrl.Result, err error, t *testing.T) {
+	return func(c client.Client, res ctrl.Result, err error, t *testing.T) {
+		if err != nil {
+			t.Fatalf("unexpected error: %s", err)
+		}
+		class := &v1beta1.GatewayClass{}
+		err = c.Get(context.TODO(), client.ObjectKey{Name: name}, class)
+		if err != nil {
+			t.Fatalf("error getting gateway class from client: %s", err)
+		}
+		if want != gatewayClassIsAccepted(class) {
+			t.Fatalf("controller ignored or not accepted gateway class")
+		}
+	}
+}
+
+func gatewayClassIsAccepted(gatewayClass *v1beta1.GatewayClass) bool {
+	acceptedCondition := conditions.GetConditionByType(gatewayClass.Status.Conditions, string(v1beta1.GatewayConditionAccepted))
+	return (acceptedCondition != nil && acceptedCondition.Status == v1.ConditionTrue)
+}
+
+func buildProgrammedStatus(gatewayStatus v1beta1.GatewayStatus, generation int64, placed []string, programmedStatus v1.ConditionStatus, err error) []v1.Condition {
+	errorMsg := ""
+	if err != nil {
+		errorMsg = err.Error()
+	}
+	message := "waiting for gateway to placed on clusters %v %s"
+	if programmedStatus == v1.ConditionTrue {
+		message = "gateway placed on clusters %v %s"
+	}
+	if programmedStatus == v1.ConditionFalse {
+		message = "gateway failed to be placed on all clusters %v error %s"
+	}
+	if programmedStatus == v1.ConditionUnknown {
+		message = "current state of the gateway is unknown error %s"
+	}
+	return conditions.SetCondition(gatewayStatus.Conditions, generation, string(v1beta1.GatewayConditionProgrammed), programmedStatus, string(v1beta1.GatewayReasonProgrammed), fmt.Sprintf(message, placed, errorMsg))
```

Comment:
Why is this here? Is this not code that belongs in the pkg?

Reply:
reusing one from the gateway package